Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=505872
Description: Define _GNU_SOURCE in translation units that need struct ucred.

diff -Nur -x '*.orig' -x '*~' gamin-0.1.9/libgamin/gam_api.c gamin-0.1.9.new/libgamin/gam_api.c
--- gamin-0.1.9/libgamin/gam_api.c	2007-07-04 14:36:48.000000000 +0100
+++ gamin-0.1.9.new/libgamin/gam_api.c	2008-11-16 08:56:18.000000000 +0000
@@ -3,6 +3,7 @@
  */
 
 #include "config.h"
+#define _GNU_SOURCE
 #include <pwd.h>
 #include <stdlib.h>
 #include <stdio.h>
diff -Nur -x '*.orig' -x '*~' gamin-0.1.9/server/gam_channel.c gamin-0.1.9.new/server/gam_channel.c
--- gamin-0.1.9/server/gam_channel.c	2007-07-04 14:36:49.000000000 +0100
+++ gamin-0.1.9.new/server/gam_channel.c	2008-11-16 08:56:23.000000000 +0000
@@ -1,4 +1,5 @@
 #include "server_config.h"
+#define _GNU_SOURCE
 #include <stdlib.h>
 #include <unistd.h>
 #include <errno.h>
